### ISSUE TITLE
Ruby 2.7.6, unable to install ddtrace 1.6.1 on Ubuntu 20.04.5   LTS (Github Action)

### ISSUE THREAD

**Issue by rud** [DataDog/dd-trace-rb#2388]
Hi maintainers,

I've tried to update a `ddtrace` dependency from 1.5.2 to 1.6.1, just to keep up. It used to install without any specific configuration, but now it fails.

Docker image: https://github.com/actions/runner-images/blob/ubuntu20/20221027.1/images/linux/Ubuntu2004-Readme.md

Bundler version: 2.3.25

In the log from the failed build I see:

```
  Installing libddwaf 1.5.1.0.0 (x86_64-linux)
  Installing libdatadog 0.9.0.1.0 (x86_64-linux)
  Fetching ddtrace 1.6.1
  Installing ddtrace 1.6.1 with native extensions
  Gem::Ext::BuildError: ERROR: Failed to build gem native extension.
  
  current directory:
  /home/runner/work/project/vendor/bundle/ruby/2.7.0/gems/ddtrace-1.6.1/ext/ddtrace_profiling_native_extension
  /opt/hostedtoolcache/Ruby/2.7.6/x64/bin/ruby -I
  /opt/hostedtoolcache/Ruby/2.7.6/x64/lib/ruby/2.7.0 -r
  ./siteconf20221121-4424-1wymgar.rb extconf.rb
  
  +------------------------------------------------------------------------------+
  | ** Preparing to build the ddtrace profiling native extension... **           |
  |                                                                              |
  | If you run into any failures during this step, you can set the               |
  | `DD_PROFILING_NO_EXTENSION` environment variable to `true` e.g.              |
  | `$ DD_PROFILING_NO_EXTENSION=true bundle install` to skip this step.         |
  |                                                                              |
  | If you disable this extension, the Datadog Continuous Profiler will          |
  | not be available, but all other ddtrace features will work fine!             |
  |                                                                              |
  | If you needed to use this, please tell us why on                             |
  | <https://github.com/DataDog/dd-trace-rb/issues/new> so we can fix it :)      |
  |                                                                              |
  | Thanks for using ddtrace! You rock!                                          |
  +------------------------------------------------------------------------------+
  
  extconf.rb:204: warning: already initialized constant
  MakeMakefile::COMMON_HEADERS
  /opt/hostedtoolcache/Ruby/2.7.6/x64/lib/ruby/2.7.0/mkmf.rb:2684: warning:
  previous definition of COMMON_HEADERS was here
  checking for RUBY_MJIT_H in rb_mjit_min_header-2.7.6.h... yes
  extconf.rb:208: warning: already initialized constant
  MakeMakefile::COMMON_HEADERS
  extconf.rb:204: warning: previous definition of COMMON_HEADERS was here
  creating extconf.h
  creating Makefile
  
  current directory:
  /home/runner/work/project/vendor/bundle/ruby/2.7.0/gems/ddtrace-1.6.1/ext/ddtrace_profiling_native_extension
  make "DESTDIR=" clean
  
  current directory:
  /home/runner/work/project/vendor/bundle/ruby/2.7.0/gems/ddtrace-1.6.1/ext/ddtrace_profiling_native_extension
  make "DESTDIR="
  compiling clock_id_from_pthread.c
  compiling clock_id_noop.c
  compiling collectors_cpu_and_wall_time.c
  compiling collectors_cpu_and_wall_time_worker.c
  compiling collectors_stack.c
  collectors_stack.c: In function ‘sample_thread’:
  collectors_stack.c:151:7: error: missing initializer for field ‘start_line’ of
  ‘ddog_Function’ {aka ‘struct ddog_Function’}
  [-Werror=missing-field-initializers]
    151 |       },
        |       ^
  In file included from
  /home/runner/work/project/vendor/bundle/ruby/2.7.0/gems/libdatadog-0.9.0.1.0-x86_64-linux/vendor/libdatadog-0.9.0/x86_64-linux/libdatadog-x86_64-unknown-linux-gnu/lib/pkgconfig/../../include/datadog/profiling.h:11,
                   from libdatadog_helpers.h:3,
                   from collectors_stack.c:5:
  /home/runner/work/project/vendor/bundle/ruby/2.7.0/gems/libdatadog-0.9.0.1.0-x86_64-linux/vendor/libdatadog-0.9.0/x86_64-linux/libdatadog-x86_64-unknown-linux-gnu/lib/pkgconfig/../../include/datadog/common.h:307:11:
  note: ‘start_line’ declared here
    307 |   int64_t start_line;
        |           ^~~~~~~~~~
  collectors_stack.c: In function ‘maybe_add_placeholder_frames_omitted’:
  collectors_stack.c:307:5: error: missing initializer for field ‘start_line’ of
  ‘ddog_Function’ {aka ‘struct ddog_Function’}
  [-Werror=missing-field-initializers]
    307 |     },
        |     ^
  In file included from
  /home/runner/work/project/vendor/bundle/ruby/2.7.0/gems/libdatadog-0.9.0.1.0-x86_64-linux/vendor/libdatadog-0.9.0/x86_64-linux/libdatadog-x86_64-unknown-linux-gnu/lib/pkgconfig/../../include/datadog/profiling.h:11,
                   from libdatadog_helpers.h:3,
                   from collectors_stack.c:5:
  /home/runner/work/project/vendor/bundle/ruby/2.7.0/gems/libdatadog-0.9.0.1.0-x86_64-linux/vendor/libdatadog-0.9.0/x86_64-linux/libdatadog-x86_64-unknown-linux-gnu/lib/pkgconfig/../../include/datadog/common.h:307:11:
  note: ‘start_line’ declared here
    307 |   int64_t start_line;
        |           ^~~~~~~~~~
  collectors_stack.c: In function ‘record_placeholder_stack_in_native_code’:
  collectors_stack.c:344:5: error: missing initializer for field ‘start_line’ of
  ‘ddog_Function’ {aka ‘struct ddog_Function’}
  [-Werror=missing-field-initializers]
    344 |     },
        |     ^
  In file included from
  /home/runner/work/project/vendor/bundle/ruby/2.7.0/gems/libdatadog-0.9.0.1.0-x86_64-linux/vendor/libdatadog-0.9.0/x86_64-linux/libdatadog-x86_64-unknown-linux-gnu/lib/pkgconfig/../../include/datadog/profiling.h:11,
                   from libdatadog_helpers.h:3,
                   from collectors_stack.c:5:
  /home/runner/work/project/vendor/bundle/ruby/2.7.0/gems/libdatadog-0.9.0.1.0-x86_64-linux/vendor/libdatadog-0.9.0/x86_64-linux/libdatadog-x86_64-unknown-linux-gnu/lib/pkgconfig/../../include/datadog/common.h:307:11:
  note: ‘start_line’ declared here
    307 |   int64_t start_line;
        |           ^~~~~~~~~~
  collectors_stack.c: In function ‘sampling_buffer_new’:
  collectors_stack.c:376:5: error: missing initializer for field ‘is_folded’ of
  ‘ddog_Location’ {aka ‘struct ddog_Location’}
  [-Werror=missing-field-initializers]
  376 |     buffer->locations[i] = (ddog_Location) {.lines = (ddog_Slice_line)
  {.ptr = &buffer->lines[i], .len = 1}};
        |     ^~~~~~
  In file included from
  /home/runner/work/project/vendor/bundle/ruby/2.7.0/gems/libdatadog-0.9.0.1.0-x86_64-linux/vendor/libdatadog-0.9.0/x86_64-linux/libdatadog-x86_64-unknown-linux-gnu/lib/pkgconfig/../../include/datadog/profiling.h:11,
                   from libdatadog_helpers.h:3,
                   from collectors_stack.c:5:
  /home/runner/work/project/vendor/bundle/ruby/2.7.0/gems/libdatadog-0.9.0.1.0-x86_64-linux/vendor/libdatadog-0.9.0/x86_64-linux/libdatadog-x86_64-unknown-linux-gnu/lib/pkgconfig/../../include/datadog/common.h:360:8:
  note: ‘is_folded’ declared here
    360 |   bool is_folded;
        |        ^~~~~~~~~
  cc1: all warnings being treated as errors
  make: *** [Makefile:245: collectors_stack.o] Error 1
  
  make failed, exit code 2
```

What have I missed? The install went fine on my local machine.

**Comment by TonyCTHsu** [comment]
:wave: @rud , thanks for reporting and sorry that you ran into this. We've prepared a [fix](https://github.com/DataDog/dd-trace-rb/pull/2378) for this issue, and it will be included in the next stable release. As a workaround to unblock you, we suggest setting the `DD_PROFILING_NO_EXTENSION=true` environment variable, which will skip the compilation entirely.

**Comment by rud** [comment]
Awesome, @TonyCTHsu. Thank you for the fast turn-around.

**Comment by ivoanjo** [comment]
Thanks for the patience @rud! The fix has now been released as part of [version 1.7.0](https://github.com/DataDog/dd-trace-rb/releases/tag/v1.7.0).
